lua/nvim-treesitter/ts_utils.lua

```diff
@@ -2,35 +2,21 @@ local api = vim.api
 
 local parsers = require "nvim-treesitter.parsers"
 local utils = require "nvim-treesitter.utils"
+local ts_query = vim.treesitter.query
 
 local M = {}
 
 --- Gets the actual text content of a node
+-- @deprecated Use vim.treesitter.query.get_node_text
 -- @param node the node to get the text from
 -- @param bufnr the buffer containing the node
 -- @return list of lines of text of the node
 function M.get_node_text(node, bufnr)
-  local bufnr = bufnr or api.nvim_get_current_buf()
-  if not node then
-    return {}
-  end
-
-  -- We have to remember that end_col is end-exclusive
-  local start_row, start_col, end_row, end_col = M.get_node_range(node)
-
-  if start_row ~= end_row then
-    local lines = api.nvim_buf_get_lines(bufnr, start_row, end_row + 1, false)
-    lines[1] = string.sub(lines[1], start_col + 1)
-    -- end_row might be just after the last line. In this case the last line is not truncated.
-    if #lines == end_row - start_row + 1 then
-      lines[#lines] = string.sub(lines[#lines], 1, end_col)
-    end
-    return lines
-  else
-    local line = api.nvim_buf_get_lines(bufnr, start_row, start_row + 1, false)[1]
-    -- If line is nil then the line is empty
-    return line and { string.sub(line, start_col + 1, end_col) } or {}
-  end
+  require("nvim-treesitter.utils").notify(
```

Comment:
Use  `vim.notify`, since we're deprecating.

Reply:
this wrapper is just adding `nvim-treesitter` as the source for vim.notify

Reply:
And I'm saying "no wrapper needed" :)

Reply:
Sorry, should have been clearer: We want to use `notify_once` for deprecation messages to avoid spamming them continuously.

Reply:
Is it even safe to display messages in contexts where `get_node_text` is used (e.g. highlighting queries, at least not in our predicates)

Reply:
Not sure; `notify_once` is meant for exactly that purpose (avoid spamming in hot loops), but there may still be a performance hit.

Since right now were ripping the bandaids off anyway, I'd just hard-deprecate it and not worry about it; there will be a bunch of breaking commits at the same time, anyway.